libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2608,10 +2612,12 @@ vector<VariableDeclaration const*> SMTEncoder::modifiersVariables(FunctionDefini
 			continue;
 
 		visited.insert(modifier);
-		if (modifier->isImplemented())
+		solAssert(_contract, "No contract context provided for modifier analysis!");
+		auto const& actualModifier = modifier->resolveVirtual(*_contract);
```

Comment:
I'm wondering now if `SMTEncoder::inlineModifierInvocation` is buggy because of virtual modifiers

Reply:
I don't see a problem right now, why do you think it could be buggy?